File: its/ruling/src/test/resources/sources/ruby/S1862.rb
if param == 1
  openWindow()
elsif param == 2
  closeWindow()
elsif param == 1  # Noncompliant
  moveWindowToTheBackground()
end

case i
  when 1
    # ...
  when 3
    # ...
  when 1  # Noncompliant
    # ...
  else
    # ...
end

if param == 1
  openWindow()
elsif param == 2
  closeWindow()
elsif param == 3
  moveWindowToTheBackground()
end

case i
  when 1
    # ...
  when 3
    # ...
  else
    # ...
end

File: its/ruling/src/test/resources/sources/scala/S1862.scala
// copyright
object Code {
  def method(param: Int): Int = {
    if (param == 1) {
      openWindow
    } else if (param == 2) {
      closeWindow
    } else if (param == 1) { // Noncompliant
      moveWindowToTheBackground
    } else {
      moveWindow
    }
    param match {
      case 1 =>
        7
      case 3 =>
        13
      case 1 => // Noncompliant
        19
      case _ =>
        23
    }
  }
}
